pkg/reconciler/event_test.go

```diff
@@ -127,43 +122,70 @@ func TestEmitEvent(t *testing.T) {
 			Status:  corev1.ConditionFalse,
 			Message: "really bad",
 		},
-		expectEvent:   true,
-		expectedEvent: "Warning Failed really bad",
+		wantEvent: "Warning Failed really bad",
 	}, {
 		name:   "nil to false",
 		before: nil,
 		after: &apis.Condition{
 			Type:   apis.ConditionSucceeded,
 			Status: corev1.ConditionFalse,
 		},
-		expectEvent:   true,
-		expectedEvent: "Warning Failed ",
+		wantEvent: "Warning Failed ",
 	}}
 
 	for _, ts := range testcases {
 		fr := record.NewFakeRecorder(1)
 		tr := &corev1.Pod{}
 		EmitEvent(fr, ts.before, ts.after, tr)
-		timer := time.NewTimer(1 * time.Second)
 
-		select {
-		case event := <-fr.Events:
-			if event == "" {
-				// The fake recorder reported empty, it should not happen
-				t.Fatalf("Expected event but got empty for %s", ts.name)
-			}
-			if !ts.expectEvent {
-				// The fake recorder reported an event which we did not expect
-				t.Errorf("Unxpected event \"%s\" but got one for %s", event, ts.name)
-			}
-			if !(event == ts.expectedEvent) {
-				t.Errorf("Expected event \"%s\" but got \"%s\" instead for %s", ts.expectedEvent, event, ts.name)
-			}
-		case <-timer.C:
-			if ts.expectEvent {
-				// The fake recorder did not report, the timer timeout expired
-				t.Errorf("Expected event but got none for %s", ts.name)
-			}
+		err := checkEvents(fr, ts.name, ts.wantEvent)
+		if err != nil {
+			t.Errorf(err.Error())
 		}
 	}
 }
+
+func TestEmitErrorEvent(t *testing.T) {
+	testcases := []struct {
+		name      string
+		err       error
+		wantEvent string
+	}{{
+		name:      "with error",
+		err:       errors.New("something went wrong"),
+		wantEvent: "Warning Error something went wrong",
+	}, {
+		name:      "without error",
+		err:       nil,
+		wantEvent: "",
+	}}
+
+	for _, ts := range testcases {
+		fr := record.NewFakeRecorder(1)
+		tr := &corev1.Pod{}
+		EmitErrorEvent(fr, ts.err, tr)
+
+		err := checkEvents(fr, ts.name, ts.wantEvent)
+		if err != nil {
+			t.Errorf(err.Error())
```

Comment:
might be nice to add some context, e.g. "expected error when blah blah"

Reply:
The context is added in the error by `checkEvents` already.
I could add `unexpected error when checking for events: %#v", err.Error()` but I'm not sure it would add much.